internal/proxy/plan_parser.go

```diff
@@ -517,6 +679,12 @@ func (pc *parserContext) handleMultiCmpExpr(node *ant_ast.BinaryNode) (*planpb.E
 }
 
 func (pc *parserContext) handleBinaryExpr(node *ant_ast.BinaryNode) (*planpb.Expr, error) {
+	_, arithExpr := node.Left.(*ant_ast.FunctionNode)
+
+	if arithExpr {
+		return pc.handleBinaryArithCmpExpr(node)
```

Comment:
It's ok that our optimizer handles the ExprTree well (optimizer transfer arithmetic-related node to FunctionNode). You can also consider `BinaryArithCmpExpr` as a special cmp expr and then handle it in cmp expr recursively.
Both these ways make sense for me.

Reply:
Ah, I decided to implement the `FunctionNode` since I find it to aptly represent what the feature wanted to accomplish, i.e., to analyze the expression as a function. Nonetheless, I will think through how to better design this and potentially implement the `BinaryArithCmpExpr`  in the next PR if it serves to be a convenient choice when expanding the functionality to support other comparison operators. Also, I remember that I ended up doing the `FunctionNode` transform because, originally, the optimizer will raise an exception if an arithmetic operator has an `IdentifierNode` as its operands. So I had to handle it at the optimizer. 😄